### DISCUSSION TITLE
Well Plot + Display Multiple 3D object in One Display

### DISCUSSION THREAD

**Discussion by kusuma2022** [pyvista/pyvista#2033] [Q&A]
Hi there,

I'm new to pyvista, 
I would like to display well with facies data in 3D view and in the same time, I would like to display other 3D object (e.g slice of 3D grid) in the same window.
Is this doable in pyvista ? How to do this? 
[well_ijk_fac_example_pyvist.xlsx](https://github.com/pyvista/pyvista/files/7877516/well_ijk_fac_example_pyvist.xlsx)

thanks

**Comment by RichardScottOZ** [comment]
Add them as separate meshes to 
p = pv.plotter()
p.add_mesh rocks
p.add_mesh depthslice


As simplified overview

**Reply by kusuma2022** [reply]
thank Richard.. do you know how to plot the well data ?

**Reply by kusuma2022** [reply]
finally I can display them. thank Richard

**Reply by kusuma2022** [reply]
@banesullivan : I have question about the well data display , since they are in grid format (IJK), can we make them as grid also? right now what I have are points display. thanks
![image](https://user-images.githubusercontent.com/97845744/151846035-1de56887-67d8-45b1-84be-76fd4f59d1cd.png)

**Reply by banesullivan** [reply]
> since they are in grid format (IJK), can we make them as grid also? right now what I have are points display

@kusuma2022, isn't this what you have in your `gridpv` object? I think you could just add that to the plotter as well

**Reply by kusuma2022** [reply]
no. the wells are not in gridpv. 
gridpv is only the slice of the 3D grid.

the borehole wells are coming from your script in other person discussion, here is the script: 
boreholdeijk = df[['j_index','i_index', 'k_index']] 
boreholefacies = df[['Facies']]
boreholes = pv.PolyData(np.c_[boreholdeijk])
boreholes['labels'] = boreholefacies

**Reply by banesullivan** [reply]
Can you share the raw data?

**Reply by kusuma2022** [reply]
@banesullivan : here is the raw data. Actually, It is 3d array numpy data but I need to reshape it into 2D array using shape[0],-1 and save using np.save text. what I understand we need to load and reshape it again using this  (load_original_arr = loaded_arr.reshape(loaded_arr.shape[0], loaded_arr.shape[1] // arr.shape[2], arr.shape[2])). 
thanks in advance
[df_fac.txt](https://github.com/pyvista/pyvista/files/8125870/df_fac.txt)

**Comment by RichardScottOZ** [comment]
You can do this sort of thing:- https://github.com/cgre-aachen/gemgis/issues/206#issuecomment-1022005426

**Reply by kusuma2022** [reply]
thank Richard..seems quite long =>
